src/test/kotlin/graphics/scenery/tests/examples/volumes/ClientApplication.kt

```diff
@@ -0,0 +1,457 @@
+            Texture.UsageType.LoadStoreImage, Texture.UsageType.Texture, Texture.UsageType.AsyncLoad), type = FloatType(), mipmap = false, normalized = false, minFilter = Texture.FilteringMode.NearestNeighbour, maxFilter = Texture.FilteringMode.NearestNeighbour)
+        val depthUpdate = UpdatableTexture.TextureUpdate(
+            UpdatableTexture.TextureExtents(0, 0, 0, 2 * numSupersegments, windowHeight, windowWidth),
+            depth.slice()
+        )
+        depthTexture.addUpdate(depthUpdate)
+
+        val numGridCells = Vector3f(vdiData.metadata.windowDimensions.x.toFloat() / 8f, vdiData.metadata.windowDimensions.y.toFloat() / 8f, numSupersegments.toFloat())
+        val accelTexture = UpdatableTexture(Vector3i(numGridCells.x.toInt(), numGridCells.y.toInt(), numGridCells.z.toInt()), channels = 1, contents = null, usageType = hashSetOf(
+            Texture.UsageType.LoadStoreImage, Texture.UsageType.Texture, Texture.UsageType.AsyncLoad), type = UnsignedIntType(), mipmap = false, normalized = true, minFilter = Texture.FilteringMode.NearestNeighbour, maxFilter = Texture.FilteringMode.NearestNeighbour)
+        val accelUpdate = UpdatableTexture.TextureUpdate(
+            UpdatableTexture.TextureExtents(0, 0, 0, windowWidth / 8, windowHeight / 8, numSupersegments),
+            accelGridBuffer
+        )
+        accelTexture.addUpdate(accelUpdate)
+
+        if (firstVDI || compute.useSecondBuffer) { //if this is the first VDI or the second buffer was being used so far
+            compute.ViewOriginal = vdiData.metadata.view
+            compute.invViewOriginal = Matrix4f(vdiData.metadata.view).invert()
+
+            compute.material().textures["InputVDI"] = colorTexture
+            compute.material().textures["DepthVDI"] = depthTexture
+            compute.material().textures["OctreeCells"] = accelTexture
+
+            logger.info("Uploading data for buffer 1")
+        } else {
+            compute.ViewOriginal2 = vdiData.metadata.view
+            compute.invViewOriginal2 = Matrix4f(vdiData.metadata.view).invert()
+
+            compute.material().textures["InputVDI2"] = colorTexture
+            compute.material().textures["DepthVDI2"] = depthTexture
+            compute.material().textures["OctreeCells2"] = accelTexture
+
+            logger.info("Uploading data for buffer 2")
+        }
+
+        while (!colorTexture.availableOnGPU() || !depthTexture.availableOnGPU()) {
+            logger.debug("Waiting for texture transfer. color: ${colorTexture.availableOnGPU()} and depth: ${depthTexture.availableOnGPU()}")
+            Thread.sleep(10)
+        }
+
+        logger.debug("Data has been detected to be uploaded to GPU")
+        newVDI = true
+
+        if (!firstVDI) {
+            compute.useSecondBuffer = !compute.useSecondBuffer
+        }
+    }
+
+    private fun receiveAndUpdateVDI(compute: VDINode) {
```

Comment:
It looks like there's a lot of functionality here that is generic, and should therefore not be part of the example, but of the VDI generation/receiving parts.